From 342f2a96b636a80e1b4dc8a16ef7a86669fe4b80 Mon Sep 17 00:00:00 2001
From: Piotr Gorski <lucjan.lucjanov@gmail.com>
Date: Wed, 22 Sep 2021 20:15:11 -0500
Subject: [PATCH 2/2] sched/alt: add missing sanity check in
 migrate_pending_tasks()

As fixed by Alfred in the link below:
https://gitlab.com/alfredchen/linux-prjc/-/issues/36#note_681549062
---
 kernel/sched/alt_core.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kernel/sched/alt_core.c b/kernel/sched/alt_core.c
index 258f11b..c1643fa 100644
--- a/kernel/sched/alt_core.c
+++ b/kernel/sched/alt_core.c
@@ -4031,6 +4031,7 @@ migrate_pending_tasks(struct rq *rq, struct rq *dest_rq, const int dest_cpu)
 		if (cpumask_test_cpu(dest_cpu, p->cpus_ptr)) {
 			__SCHED_DEQUEUE_TASK(p, rq, 0, );
 			set_task_cpu(p, dest_cpu);
+			sched_task_sanity_check(p, dest_rq);
 			__SCHED_ENQUEUE_TASK(p, dest_rq, 0);
 			nr_migrated++;
 		}
-- 
2.33.0.328.g8b7c11b866

